### PR TITLE
[JH] [AM] Add NavLink

### LINKED ISSUE #3: 3. As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
## Summary

Users need to be able to navigate between the pages of our application in order to access all of its features. There is a navigation bar in the `Layout.jsx` component, but it's not quite feature-complete: the links in the nav element don't actually allow the user to navigate from page to page.

## Acceptance criteria
- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Notes
- The [official docs for the `NavLink` component](https://reactrouter.com/en/6.14.2/components/nav-link) will come in very handy!
- While a standard anchor element (`<a>`) _could_ allow a user to navigate from page to page, the `NavLink` component adds additional behavior that is essential to our app functioning correctly!

### PR DESCRIPTION
**Pull request by hsiangj**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

Changed anchor element to NavLink component from react-router-dom. This allows for automatic active link functionality. Also added 'to' attribute for links to function as expected. 

## Related Issue

Closes #3 

## Acceptance Criteria

- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before

Screen Shot 2024-02-05 at 10 30 01 AM

### After

Screen Shot 2024-02-05 at 10 29 15 AM
Screen Shot 2024-02-05 at 10 29 06 AM

## Testing Steps / QA Criteria

Should be able to pull down the code and navigate to 'Home', 'List', and 'Manage-List' links respectively by clicking the hyperlink on the bottom of the page.

**NOTE:** "Add item" page is called "Manage List" in Layout.jsx

### CODE REVIEW

**Review by krsnamara — APPROVED** [review]
Looks great! nit: Remove TODO code

**Review by rmorabia — APPROVED** [review]
Looks good to me! Great job, y'all. 🥳

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { auth } from '../api/config.js';
 /**
  * TODO: The links defined in this file don't work!
  *
- * Instead of anchor element, they should use a component
+ * Instead of anchor element, they should use NavLink component
  * from `react-router-dom` to navigate to the routes
  * defined in `App.jsx`.
  */
```
Comment (louieknolle) [inline src/views/Layout.jsx:13]: Doesn't need to be now, but in general ToDo comments can be removed once the task is complete.

**Comment by louieknolle** [comment]
Ditto!

**Pull request merged**